RHNO1.py

```diff
@@ -0,0 +1,146 @@
+from Bio import SeqIO
+from Bio.SeqUtils import gc_fraction
+from abc import ABC, abstractmethod
+
+
+class FastQFilter:
+    def __init__(
+        self, input_file, output_file, min_length=0, min_quality=0, min_gc=0, max_gc=100
+    ):
+        self.input_file = input_file
+        self.output_file = output_file
+        self.min_length = min_length
+        self.min_quality = min_quality
+        self.min_gc = min_gc
+        self.max_gc = max_gc
+
+    def filter_fastq(self):
+        with open(self.output_file, "w") as output_handle:
+            for record in SeqIO.parse(self.input_file, "fastq"):
+                if self._passes_filter(record):
+                    SeqIO.write(record, output_handle, "fastq")
+
+    def _passes_filter(self, record):
+        if len(record.seq) < self.min_length:
+            return False
+
+        if min(record.letter_annotations["phred_quality"]) < self.min_quality:
+            return False
+
+        gc_content = gc_fraction(record.seq)
+        if gc_content < self.min_gc or gc_content > self.max_gc:
+            return False
+
+        return True
+
+
+class BiologicalSequence(ABC):
+    @abstractmethod
+    def __len__(self):
+        pass
+
+    @abstractmethod
+    def __getitem__(self, index):
+        pass
+
+    @abstractmethod
+    def __str__(self):
+        pass
+
+    @abstractmethod
+    def check_alphabet(self):
+        pass
+
+
+class NucleicAcidSequence(BiologicalSequence):
+    complement_dict = {"A": "", "T": "", "G": "", "C": ""}
+
+    def __init__(self, sequence):
+        self.sequence = sequence
+
+    def __len__(self):
+        return len(self.sequence)
+
+    def __getitem__(self, index):
+        return self.sequence[index]
+
+    def __str__(self):
+        return self.sequence
+
+    def check_alphabet(self):
+        valid_alphabet = set("ATGC")
+        return set(self.sequence) <= valid_alphabet
+
+    def complement(self):
+        complemented_sequence = [
+            self.complement_dict.get(base, base) for base in self.sequence
+        ]
+        return self._create_instance("".join(complemented_sequence))
+
+    def _create_instance(self, sequence):
+        return self.__class__(sequence)
+
+    def gc_content(self):
+        gc_count = sum(base in "GCgc" for base in self.sequence)
+        return gc_count / len(self.sequence)
+
+
+class DNASequence(NucleicAcidSequence):
+    complement_dict = {"A": "T", "T": "A", "G": "C", "C": "G"}
```

Comment:
Вероятно этот комментарий супер вкусовщина. Более приятно читается словарь, когда он построчный. Но это сугубо вкусовщина 
```suggestion
    complement_dict = {"A": "T",
                       "T": "A", 
                       "G": "C", 
                       "C": "G"}
```

Reply:
следующие словарики я бы также подпарвил